server/src/main/java/org/opensearch/action/ActionModule.java

```diff
@@ -471,7 +475,17 @@ public class ActionModule extends AbstractModule {
     private final ClusterSettings clusterSettings;
     private final SettingsFilter settingsFilter;
     private final List<ActionPlugin> actionPlugins;
+    // The unmodifiable map containing OpenSearch and Plugin actions
+    // This is initialized at node bootstrap and contains same-JVM actions
+    // It will be wrapped in the Dynamic Action Registry but otherwise
+    // remains unchanged from its prior purpose, and registered actions
+    // will remain accessible.
     private final Map<String, ActionHandler<?, ?>> actions;
+    // A dynamic action registry which includes the above immutable actions
+    // and also registers dynamic ExtensionActions / ExtensionTransportActions
+    // associated with remote action execution on extensions, usually in
+    // a different JVM and possibly on a different server.
+    private final DynamicActionRegistry dynamicActionRegistry;
```

Comment:
I like how you've separated out dynamic actions differently, what concerns do you see of modifying the existing actions ? I see we aren't using them anyway. 
I can think of inconsistencies with other nodes which might not have registered actions after bootstrap. 

I understand the safeguards make sense for 2.x but for 3.x Im leaning towards having one set of actions which are injected and read by the client.

Reply:
@saratvemulapalli "injected" only happens once, and the whole ActionType instance -> TransportAction class setup is tightly integrated.   The injector already has the necessary instances.   Removing them from the map wouldn't take away the singleton that is still bound to that class.    In order to truly "unregister" the things loaded in during bootstrap we need to get away from injection.

Reply:
> I can think of inconsistencies with other nodes which might not have registered actions after bootstrap.

This is actually a bigger issue that I hadn't designed for.  What is the easiest existing method for replicating that?   I designed this registry after named registry, does that replicate across nodes?

Reply:
OK..... some follow-up comments:
1. Conceptually this just boils down to an immutable map and a mutable one, plus some methods to access it
2. I will do renaming to make it more generic describing the above
3. There is some point that `client.execute()` on the NodeClient needs access to both maps.   We have the choice of adding a separate map and passing both to NodeClient (my original PR before refactoring in response to @peternied comments) or combining them here inside this module.   Or combining them somewhere else in a whole new module.   I think here is the simplest approach, particularly since they are all "actions".

Reply:
Makes sense thanks @dbwiddis !